src/metabase/models/database.clj

```diff
@@ -86,9 +86,7 @@
 (defn- pre-delete [{id :id, driver :engine, :as database}]
   (unschedule-tasks! database)
   (db/execute! {:delete-from (db/resolve-model 'Permissions)
-                :where       [:or
-                              [:like :object (str (perms/data-perms-path id) "%")]
-                              [:= :object (perms/database-block-perms-path id)]]})
+                :where       [:like :object (str "%" (perms/data-perms-path id) "%")]})
```

Comment:
This ensures that download perms for a DB are deleted when a DB is deleted. We should just be able to safely delete all perm paths that include the data perms path for the DB within them.